Kotlin/app/src/main/java/com/taboola/kotlin/examples/screens/classic/MetaUICustomizationClassicUnitFragment.kt

```diff
@@ -0,0 +1,158 @@
+    ): View? {
+        Taboola.init(TBLPublisherInfo(MetaConst.META_PUBLISHER_NAME))
+        val rootView = inflater.inflate(R.layout.fragment_meta_ad_classic_unit, null)
+        val adContainerTop: NativeAdLayout = rootView.findViewById(R.id.native_ad_container_top)
+        Taboola.setGlobalExtraProperties(object : HashMap<String?, String?>() {
+            init {
+
+                val isDarkTheme = isDarkTheme(requireActivity())
+                put(
+                    MetaConst.AUDIENCE_NETWORK_APPLICATION_ID_KEY, MetaConst.AUDIENCE_NETWORK_APP_ID
+                )
+                put(MetaConst.ENABLE_META_DEMAND_DEBUG_KEY, "true")
+
+                //if the device is in dark mode, set the dark mode flag
+                if (isDarkTheme) {
+                    put(DARK_MODE, "true")
+                }
+            }
+        })
+        setupAndLoadTaboolaAd(adContainerTop)
+        return rootView
+    }
+
+    // check OS dark mode
+    fun isDarkTheme(activity: Activity): Boolean {
+        return activity.resources.configuration.uiMode and
+                Configuration.UI_MODE_NIGHT_MASK == Configuration.UI_MODE_NIGHT_YES
+    }
+
+
+    private fun setupAndLoadTaboolaAd(adContainer: NativeAdLayout) {
+        val widgetProperties = PlacementInfo.metaWidgetProperties()
+        val feedProperties = PlacementInfo.metaFeedProperties()
+        val tblClassicPage = Taboola.getClassicPage(feedProperties.pageUrl, feedProperties.pageType)
+        val metaPlacementProperties =
+            MetaPlacementProperties(widgetProperties.placementName, widgetProperties.mode)
+        val tblMetaClassicUnit = tblClassicPage.buildWithMeta(context,
+            feedProperties.placementName,
+            feedProperties.mode,
+            TBL_PLACEMENT_TYPE.PAGE_MIDDLE,
+            metaPlacementProperties,
+            object : TBLClassicListener() {
+                override fun onItemClick(
+                    placementName: String?,
+                    itemId: String?,
+                    clickUrl: String?,
+                    isOrganic: Boolean,
+                    customData: String?
+                ): Boolean {
+                    Log.d(TAG, "onItemClick")
+                    return super.onItemClick(placementName, itemId, clickUrl, isOrganic, customData)
+                }
+
+                override fun onAdReceiveSuccess() {
+                    super.onAdReceiveSuccess()
+                    Log.d(TAG, "onAdReceiveSuccess")
+                }
+
+                override fun onAdReceiveFail(error: String?) {
+                    super.onAdReceiveFail(error)
+                    Log.d(TAG, "onAdReceiveFail $error")
+                }
+
+                override fun onResize(height: Int) {
+                    super.onResize(height)
+                    Log.d(TAG, "onResize $height")
+                }
+            })
+        tblMetaClassicUnit.setMetaAdTypeForDebug(MetaConst.TEST_LAYOUT_IMAGE_LINK_TYPE)
```

Comment:
@DanielFrTB please add this comment "Force the ad to be of type image_link" here as well